packages/dashboard/src/useDhId.ts

```diff
@@ -32,9 +35,10 @@ export function useDhId(): string {
   }
 
   if (dhId == null) {
-    throw new Error(
-      `useDhId must be used within a DhIdContext provider if there is no ${DH_ID_PROP} prop`
+    log.warn(
+      `useDhId must be used within a DhIdContext provider if there is no ${DH_ID_PROP} prop. Defaulting to empty string.`
     );
+    return '';
```

Comment:
Ehhh I don't like returning an empty string either - should return a `string | null`, or even two signatures with a param e.g. `useDhId(throwOnNull: false): string | null` and `useDhId(throwOnNull: true): string`

In `useDashboardColumnFilters` we shouldn't be emitting events with a blank or null `panelId` I don't think.

Reply:
The original intent was to throw always on no ID, but that breaks dh.ui without the update to dh.ui to pass through the panel ID

I've just switched to `string | null` return for now. Although it's annoying using the branded type for `FilterColumnSourceId` because I have to cast when calling `useDhId` and changing the return type to `string | null` didn't cause the original cast to fail.

I am in favor of removing the branded type for `FilterColumnSourceId`. I don't think it's somewhere we will commonly mix up ID types and they're both strings at the end of the day. The filter doesn't care if it's a panel or widget ID, it just needs a unique identifier